src/reuse/_util.py

```diff
@@ -172,6 +184,68 @@ def _copyright_from_dep5(path: PathLike, copyright: Copyright) -> SpdxInfo:
     )
 
 
+def _parse_copyright_year(year: str) -> list:
+    """Parse copyright years and retun list."""
+    if not year:
+        ret = []
+    if re.match(r"\d{4}$", year):
+        ret = [int(year)]
+    if re.match(r"\d{4} - \d{4}$", year):
+        ret = [int(year[:4]), int(year[-4:])]
+    return ret
+
+
+def _merge_copyright_lines(copyright_lines: Set[str]) -> Set[str]:
+    """Parse all copyright lines and merge identical statements making years
+    into a range.
+    If a same statement uses multiple prefixes, use only the most frequent one.
+    """
+    copyright_in = []
+    for line in copyright_lines:
+        for pattern in _COPYRIGHT_PATTERNS:
+            match = pattern.search(line)
+            if match is not None:
+                copyright_in.append(
+                    {
+                        "statement": match.groupdict()["statement"],
+                        "year": _parse_copyright_year(
+                            match.groupdict()["year"]
+                        ),
+                        "prefix": match.groupdict()["prefix"],
+                    }
+                )
+
+    copyright_out = []
+    for statement in {item["statement"] for item in copyright_in}:
+        copyright = [
+            item for item in copyright_in if item["statement"] == statement
+        ]
+        prefixes = [item["prefix"] for item in copyright]
+
+        # Get the style of the most common prefix
+        prefix = max(set(prefixes), key=prefixes.count)
+        style = "spdx"
+        for sty in _COPYRIGHT_STYLES:
+            if prefix == _COPYRIGHT_STYLES[sty]:
+                style = sty
+                break
```

Comment:
This is a note-worthy design-decision.

**Positive:**
- It tries to match the most common pattern in the file
- It works well in the way the code is setup up, with a per-file code path
- It doesn't require explicitly preferring a style
- It works really well if a project has a certain preferred style that is applied consistently

**Negative:**
- It has no guarantee on style consistency across the codebase, it could turn out differently for different files.
- This style check is done across all copyright annotations, not on a per-copyrightholder base.

Personally I think I would have chosen a different solution by requiring users to specify a preferred style and otherwise default to SPDX style.

Reply:
> This is a note-worthy design-decision.
> 
> **Positive:**
> 
> * It tries to match the most common pattern in the file
> * It works well in the way the code is setup up, with a per-file code path
> * It doesn't require explicitly preferring a style
> * It works really well if a project has a certain preferred style that is applied consistently
> 
I agree with your point, and I think having a consistent code base to work on might've biased me a little.

> **Negative:**
> 
> * It has no guarantee on style consistency across the codebase, it could turn out differently for different files.

My intention was to minimize the delta that the merge would generate.

I think adding an optional preferred style would be easy enough. Maybe something like this: 

```diff
diff --git a/src/reuse/_util.py b/src/reuse/_util.py
index 448565ad1411..f05e5d5e047f 100644
--- a/src/reuse/_util.py
+++ b/src/reuse/_util.py
@@ -74,6 +74,14 @@ _COPYRIGHT_STYLES = {
     "symbol": "©",
 }
 
+_PREFIX_TO_STYLE = {
+    "SPDX-FileCopyrightText:": "spdx",
+    "Copyright": "string",
+    "Copyright (C)": "string-c",
+    "Copyright ©": "string-symbol",
+    "©": "symbol",
+}
+
 # Amount of bytes that we assume will be big enough to contain the entire
 # comment header (including SPDX tags), so that we don't need to read the
 # entire file.
@@ -222,13 +230,12 @@ def _merge_copyright_lines(copyright_lines: Set[str]) -> Set[str]:
         ]
         prefixes = [item["prefix"] for item in copyright]
 
-        # Get the style of the most common prefix
-        prefix = max(set(prefixes), key=prefixes.count)
-        style = "spdx"
-        for sty in _COPYRIGHT_STYLES:
-            if prefix == _COPYRIGHT_STYLES[sty]:
-                style = sty
-                break
+        if preferred_style:
+            style = preferred_style
+        else:
+            # Get the style of the most common prefix
+            prefix = max(set(prefixes), key=prefixes.count)
+            style = _PREFIX_TO_STYLE.get(prefix, 'spdx')
 
         # get year range if any
         years = []
```

Getting the `preferred_style` all the way here might require more thought.
At first glance, adding a `preferred_style` to `SpdxInfo` doesn't seem like the right way to go, and I don't really like adding a `preferred_style` parameter to all these functions. Any suggestions?

> * This style check is done across all copyright annotations, not on a per-copyrightholder base.

If I'm not mistaken, the style is extracted per statement witch would be the copyright holder.

Maybe copyright holder is a better name for that field? I don't know which is most common across the  project.
 
> Personally I think I would have chosen a different solution by requiring users to specify a preferred style and otherwise default to SPDX style.

Reply:
I agree with Nico's comments, but I think it would be fine to not guarantee a consistent style across a codebase with this as it would mean there were different styles to begin with.

Reply:
I'll resolve this, since it seems resolved to me. Feel free to unresolve.